lib/gpg.js

```diff
@@ -24,262 +19,200 @@ var spawn = require('child_process').spawn
    * @api public
    */
 
   decryptToFile: function(options, fn){
-    options = options || {};
-
-    if (typeof options.source === 'undefined'){
-      return fn.call(null, new Error('Missing \'source\' option'));
-    } else if (typeof options.dest === 'undefined'){
-      return fn.call(null, new Error('Missing \'dest\' option'));
-    }
-
-    exists(options.source, function(e){
-      if (e){
-        var buffer = new Buffer('')
-          , fileStream = stream(options.source)
-          , destStream = writeStream(options.dest)
-          , gpg = spawn('gpg', ['--decrypt']);
-
-        gpg.on('close', function (code){
-          fn.call(null, null);
-        });
-
-        pump(fileStream, gpg.stdin);
-        pump(gpg.stdout, destStream);
-      } else {
-        fn.call(null, new Error(options.source + ' does not exist'));
-      }
-    });
+    spawnGPG.streaming(options, ['--decrypt'], fn);
   },
 
   /**
    * Clearsign `str` and pass the signed message to the callback `fn`.
    *
-   * @param {String|Buffer} str
-   * @param {Array} args Optional additional arguments to pass to gpg.
-   * @param {Function} callback containing the signed message Buffer.
+   * @param {String|Buffer} str  String to clearsign.
+   * @param {Array}         [args] Array of additonal gpg arguments.
+   * @param {Function}      fn   Callback containing the signed message Buffer.
    * @api public
    */
 
   clearsign: function(str, args, fn){
-    var defaultArgs = globalArgs.concat(['--clearsign']);
-    if (typeof args === 'function'){
+    spawnGPG(str, ['--clearsign'], args, fn);
+  },
+
+  /**
+   * Verify `str` and pass the output to the callback `fn`.
+   *
+   * @param {String|Buffer} str    Signature to verify.
+   * @param {Array}         [args] Array of additonal gpg arguments.
+   * @param {Function}      [fn]   Callback containing the signed message Buffer.
+   * @api public
+   */
+  verifySignature: function(str, args, fn){
+    // Set logger fd, verify otherwise outputs to stderr for whatever reason
+    var defaultArgs = ['--logger-fd', '1', '--verify'];
+    spawnGPG(str, defaultArgs, args, fn);
+  },
+
+  /**
+   * Add a key to the keychain by filename.
+   * @param {String}  fileName  Key filename.
+   * @param {Array}   [args]    Array of additonal gpg arguments.
+   * @param {Function} [fn]     Callback containing the signed message Buffer.
+   * @api public
+   */
+  importKeyFromFile: function(fileName, args, fn){
+    if (typeof args === 'function') {
       fn = args;
-      args = defaultArgs;
-    } else {
-      args = defaultArgs.concat(args);
+      args = [];
     }
 
-    var buffers = []
-      , buffersLength = 0
-      , error = ''
-      , gpg = spawn('gpg', args);
-
-    gpg.stdout.on('data', function(buf){
-      buffers.push(buf);
-      buffersLength += buf.length;
+    fs.readFile(fileName, function(readErr, str) {
+      if (readErr) return fn(readErr);
+      GPG.importKey(str, args, fn);
```

Comment:
I think for consistency with the rest of the file this should be `self.importKey`